src/components/Match/renderMatch.js

```diff
@@ -92,66 +96,100 @@ function generateTeamfights(match) {
   return (match.teamfights || []).map(tf => computeTfData(tf));
 }
 
+// create a detailed history of each wards
+function generateWardLog(match) {
+  const computeWardData = (player, i) => {
+    const sameWard = _.curry((w1, w2) => w1.ehandle === w2.ehandle);
+
+    // let's zip the *_log and the *_left log in a 2-tuples
+    const extractWardLog = (type, enteredLog, leftLog) =>
+       enteredLog.map((e) => {
+         const wards = [e, leftLog.find(sameWard(e))];
+         return {
+           player: i,
+           key: wards[0].ehandle,
+           type,
+           entered: wards[0],
+           left: wards[1],
+         };
+       })
+    ;
+
+    const observers = extractWardLog('observer', player.obs_log, player.obs_left_log);
+    const sentries = extractWardLog('sentry', player.sen_log, player.sen_left_log);
+    return _.concat(observers, sentries);
+  };
+
+  const imap = _.map.convert({ cap: false }); // cap: false to keep the index
+  const wardLog = _.flow(
+    imap(computeWardData),
+    _.flatten,
+    _.sortBy(xs => xs.entered.time),
+    imap((x, i) => ({ ...x, key: i })),
+  );
```

Comment:
If this can avoid using lodash, I would prefer that, but I don't know exactly what 'convert' is doing here. It seems weird to me that we would need to flatten this array one level, but I don't know what the data looks like.